rct229/rules/section5/section5rule45.py

```diff
@@ -0,0 +1,75 @@
+from rct229.ruleset_functions.compare_schedules import (compare_schedules, LEAP_YEAR_HOURS, REGULAR_YEAR_HOURS)
+
+
+class Section5Rule45(RuleDefinitionListIndexedBase):
+    """Rule 45 of ASHRAE 90.1-2019 Appendix G Section 5 (Envelope)"""
+
+    def __init__(self):
+        super(Section5Rule45, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, True, True),
+            each_rule=Section5Rule45.ZoneRule(),
+            index_rmr="baseline",
+            id="5-45",
+            description="The infiltration schedules are the same in the proposed RMR as in the baseline RMR.",
+            list_path="ruleset_model_instances[0].buildings[*].building_segments[*].zones[*]",
+            required_fields={
+                "$": ["calendar"],
+                "calendar": ["is_leap_year"]
+            }
+        )
+
+    def create_data(self, context, data=None):
+        rmr_proposed = context.proposed
+        return {"is_leap_year": rmr_proposed["calendar"]["is_leap_year"]}
+
+    def list_filter(self, context_item, data=None):
+        # baseline should match to proposed
+        zone_p = context_item.proposed
+        return (
+                zone_p.get("infiltration") is not None
+        )
+
+    class ZoneRule(RuleDefinitionBase):
+        def __init__(self):
+            super(Section5Rule45.ZoneRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, True),
+                required_fields={
+                    "infiltration": ["multiplier_schedule"]
+                },
+            )
+
+        def get_calc_vals(self, context, data=None):
+            zone_b = context.baseline
+            zone_p = context.proposed
+
+            compare_schedules_result_dictionary = {
+                "total_hours_compared": 0.0,
+                "total_hours_match": 0.0,
+                "eflh_difference": 0.0
+            }
+            mask_schedule = [1.0] * (LEAP_YEAR_HOURS if data["is_leap_year"] else REGULAR_YEAR_HOURS)
+
+            infiltration_p = zone_p["infiltration"]
+            if zone_b:
+                # make sure matched baseline zone is not None (required_field does not check None)
+                infiltration_b = zone_b.get["infiltration"]
+                compare_schedules_result_dictionary = compare_schedules(infiltration_b["multiplier_schedule"], infiltration_p["multiplier_schedule"], mask_schedule, 1.0, data["is_leap_year"])
+
+            return {
+                "total_hours_compared": compare_schedules_result_dictionary["total_hours_compared"],
+                "total_hours_match": compare_schedules_result_dictionary["total_hours_match"],
+                "eflh_difference": compare_schedules_result_dictionary["elfh_difference"]
+            }
+
+        def rule_check(self, context, calc_vals=None, data=None):
+            total_hours_compared = calc_vals["total_hours_compared"]
+            total_hours_match = calc_vals["total_hours_match"]
```

Comment:
This should be `total_hours_matched`.
I see that compare_schedules.py using "matched" in the documentation and "match" in the actual return value.  I think "matched" is more grammatically correct. "total_hours_match" would grammatically indicate a boolean value. 

While we are discussing compare_schedules.py, I see two other issues there:
- The equality check on line 63 should probably use std_equal() because of possible roundoff errors.
- This is very minor:  `total_hours_compared` and `total_hours_matched` are functionally integers so lines 48 and 51 should initialize the variable to 0 rather than 0.0.

Reply:
I will update both files.